### ISSUE TITLE
Make skeletons with mitochondria

### ISSUE THREAD

**Issue by miridyson** [connectome-neuprint/neuprint-python#35]
I would like to make skeletons of neurons that show the mitochondria as well as the synapses.

**Comment by stuarteberg** [comment]
Now that I'm thinking about this more carefully, I think the [existing `attach_synapses_to_skeleton()` function][1] should work just fine for any points  you want to attach, as long as you provide the columns `['x', 'y', 'z', 'type]`.  Here's an example:

[1]: https://connectome-neuprint.github.io/neuprint-python/docs/skeleton.html#neuprint.skeleton.attach_synapses_to_skeleton

```python
from neuprint import Client, fetch_skeleton, fetch_synapses, attach_synapses_to_skeleton, fetch_mitochondria

c = Client('neuprint.janelia.org', 'hemibrain:v1.2.1')

body = 1136399017
skeleton = fetch_skeleton(body, heal=True)
synapses = fetch_synapses(body)
mito = fetch_mitochondria(body)

mito['type'] = mito['mitoType'].map({'dark': 'mito-dark', 'medium': 'mito-medium', 'light': 'mito-light'})

syn_and_mito = pd.concat((synapses[[*'xyz', 'type']], mito[[*'xyz', 'type']]), ignore_index=True)

combined_skeleton = attach_synapses_to_skeleton(skeleton, syn_and_mito)
```

**Comment by miridyson** [comment]
Yes I see the logic.  Sorry if it's my very rusty python but I am getting
these errors:

ImportError: cannot import name 'attach_synapses_to_skeleton' from 'neuprint'

ImportError: cannot import name 'attach_synapses' from 'neuprint

**Comment by stuarteberg** [comment]
That function was [introduced very recently][1].  What version of neuprint-python are you using?

```bash
python -c 'import neuprint; print(neuprint.__version__)'
```

If you're using `conda` to manage your installation, try:

```bash
conda install -c flyem-forge 'neuprint-python>=0.4.21'
```

If you're using `pip`, try:

```bash
pip install --upgrade neuprint-python
```

[1]: https://connectome-neuprint.github.io/neuprint-python/docs/changelog.html

**Comment by stuarteberg** [comment]
My example code above is relatively simplistic and may not be suitable for plotting with `navis`.  Navis has a more sophisticated model of neuron skeletons, including the Connector concept.

Maybe @schlegelp has some advice for you...

**Comment by schlegelp** [comment]
You could piggy-back on the connector table in `navis` but that would be a bit hackish at this point. If all you want is to co-visualise the skeleton and the mitochondria, you're better off just adding them as scatter plot:

```python
>>> import navis
>>> # Import navis' wrapper for neuprint-python
>>> import navis.interfaces.neuprint as neu 
>>> c = Client('neuprint.janelia.org', 'hemibrain:v1.2.1')
>>> body = 1136399017
>>> # Fetch skeleton as navis neuron, including synapses
>>> skeleton = neu.fetch_skeletons(body, heal=True, with_synapses=True)
>>> mito = neu.fetch_mitochondria(body)
```

I am assuming you're plotting in Jupyter? If so, this will do the trick:

Plot in one-go

```python
>>> # Plot skeleton + all mitochondria as scatterplot
>>> fig = navis.plot3d([skeleton, mito[['x', 'y', 'z']].values], scatter_kws=dict(color='magenta', size=10))
```

Construct figure step-by-step (more control)

```python
>>> fig = navis.plot3d(skeleton, show=False)
>>> fig = navis.plot3d(mito.loc[mito.mitoType == 'medium', ['x', 'y', 'z']].values, scatter_kws=dict(color='magenta', size=10), fig=fig, show=False)
>>> fig = navis.plot3d(mito.loc[mito.mitoType == 'dark', ['x', 'y', 'z']].values, scatter_kws=dict(color='cyan', size=10), fig=fig, show=False)
>>> fig.show()
```

You can even pass the individual mito sizes as an array and get something like this:

<img width="657" alt="Screenshot 2022-05-19 at 22 39 44" src="https://user-images.githubusercontent.com/7161148/169409163-f9b410af-580e-4fda-ac06-f3c84a84efd4.png">

Hope this is helpful. I'll have a think about how to streamline this in the future.

**Comment by schlegelp** [comment]
For reference: if you pass an `(N, 3)` array of x/y/z coordinates to either `navis.plot2d` or `navis.plot3d` it will plot these data as scatterplot which you can fine-tune using the `scatter_kws` dictionary.

**Comment by schlegelp** [comment]
Also quick question @stuarteberg: what is `r1`, `r2`, `r3` in the mito table? And is there a way to get meshes for the mitochondria?

**Comment by stuarteberg** [comment]
@schlegelp I fit an ellipsoid to each mito, and `r1`, `r2`, and `r3` are the major, intermediate, and minor axes of that ellipsoid.  Many mito are not well approximated by an ellipsoid, but many are.  I don't know if anyone will find the ellipsoid radii useful, but it seemed like a reasonable thing to provide.

**Comment by miridyson** [comment]
Thank you!
